src/Agent.Worker/Container/DockerCommandManager.cs

```diff
@@ -100,48 +101,82 @@ public async Task<int> DockerPull(IExecutionContext context, string image)
             return await ExecuteDockerCommandAsync(context, "pull", image, context.CancellationToken);
         }
 
-        public async Task<string> DockerCreate(IExecutionContext context, string displayName, string image, List<MountVolume> mountVolumes, string network, string options, IDictionary<string, string> environment)
+        public async Task<string> DockerCreate(IExecutionContext context, ContainerInfo container)
         {
-            string dockerMountVolumesArgs = string.Empty;
-            if (mountVolumes?.Count > 0)
+            IList<string> dockerOptions = new List<string>();
+            // OPTIONS
+            dockerOptions.Add($"--name {container.ContainerDisplayName}");
+            dockerOptions.Add($"--label {DockerInstanceLabel}");
+            if (!string.IsNullOrEmpty(container.ContainerNetwork))
             {
-                foreach (var volume in mountVolumes)
-                {
-                    // replace `"` with `\"` and add `"{0}"` to all path.
-                    dockerMountVolumesArgs += $" -v \"{volume.SourceVolumePath.Replace("\"", "\\\"")}\":\"{volume.TargetVolumePath.Replace("\"", "\\\"")}\"";
-                    if (volume.ReadOnly)
-                    {
-                        dockerMountVolumesArgs += ":ro";
-                    }
-                }
+                dockerOptions.Add($"--network {container.ContainerNetwork}");
             }
-
-            string dockerEnvArgs = string.Empty;
-            if (environment?.Count > 0)
+            if (!string.IsNullOrEmpty(container.ContainerNetworkAlias))
+            {
+                dockerOptions.Add($"--network-alias {container.ContainerNetworkAlias}");
+            }
+            foreach (var port in container.PortMappings)
             {
-                foreach (var env in environment)
+                var portArg = string.Empty;
+                if (!string.IsNullOrEmpty(port.HostPort) && !string.IsNullOrEmpty(port.ContainerPort))
                 {
-                    dockerEnvArgs += $" -e \"{env.Key}={env.Value.Replace("\"", "\\\"")}\"";
+                    portArg = $"-p {port.HostPort}:{port.ContainerPort}";
+
+                }
+                else if (string.IsNullOrEmpty(port.HostPort) && !string.IsNullOrEmpty(port.ContainerPort))
```

Comment:
`else if (!string.IsNullOrEmpty(port.ContainerPort))` ?

Reply:
Are you suggesting this as an additional check? ContainerPort cant be/shouldnt be null. But it is valid to have a ContainerPort but no HostPort (port is assigned dynamically by host at container-start-time)

Reply:
```
if (!string.IsNullOrEmpty(port.HostPort) && !string.IsNullOrEmpty(port.ContainerPort))
{}
else if (!string.IsNullOrEmpty(port.ContainerPort))
{}
```

Reply:
Makes sense now. Changed